### PR TITLE
New rule: no-link-to-positional-params

### PR DESCRIPTION
**Pull request by scalvert**
Implements #1764. Messages match the spec in https://github.com/emberjs/rfcs/blob/master/text/0698-deprecate-link-to-positional-arguments.md.

This rule will be added to `recommended` in `next` for v3.0

### CODE REVIEW

**Comment by bmish** [comment]
Typically we use `no-` or `require-` prefixes in rule names, i.e. `no-link-to-positional-params`.

**Comment by scalvert** [comment]
@bmish sure, but there's other rules that are named `deprecated-*`. I was following that convention.

**Comment by bmish** [comment]
The `deprecated-` naming convention is something I've only ever seen used by a few old ember-template-lint rules. I generally try to follow naming conventions found in eslint and common eslint plugins instead.

**Comment by scalvert** [comment]
We should probably document that then. That would save folks time when they're contributing.

File: test/unit/rules/no-link-to-positional-params-test.js [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+  good: [
+    '<LinkTo @route="about">About Us</LinkTo>',
+    '<LinkTo @route="post" @model={{@post}}>Read {{@post.title}}...</LinkTo>',
+    `<LinkTo @route="post.comment" @models={{array post comment}}>
+        Comment by {{comment.author.name}} on {{comment.date}}
+      </LinkTo>`,
+    `<LinkTo @route="posts" @query={{hash direction="desc" showArchived=false}}>
+        Recent Posts
+      </LinkTo>`,
+  ],
```
Comment (rwjblue) [inline test/unit/rules/no-link-to-positional-params-test.js:19]: Need some good examples using `{{link-to`

**Comment by rwjblue** [comment]
LGTM once we have more curly example tests.

File: test/unit/rules/no-link-to-positional-params-test.js (hunk 1 above)
Comment (scalvert) [reply test/unit/rules/no-link-to-positional-params-test.js]: Done

**Review by rwjblue — APPROVED** [review]

**Pull request merged**